site/blog/2022/deploy-ghost-ec2.mdx

```diff
@@ -0,0 +1,158 @@
+---
+slug: deploy-ghost-ec2
+title: Deploy Ghost in EC2 with one click
+date: 2022-11-23
+authors: [yrobla]
+---
+
+# Deploy <a href="https://ghost.org/">Ghost</a> to AWS with one click
+
+Ghost is a powerful app for new-media creators to publish, share, and grow a business around their content. It comes with modern tools to build a website, publish content, send newsletters & offer paid
+subscriptions to members.
+
+Ghost is a software as a service product, but it is built totally on open source and offers a self-hosted, self-serve version. This version can be hosted on AWS using several approaches, and one of them.
+
+Starting an EC2 instance and manually triggering the installation commands is a procedure that can be done manually, but there are also another dependencies involved, such as the creation of proper security groups.
+It is time-consuming task, specially if you need to install several instances. This article will offer a way to install Ghost with one-click on IaSQL using an EC2 instance on Ubuntu as highlighted <a href="https://ghost.org/docs/install/ubuntu/">here</a>.
+
+_This approach uses <a href="https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/user-data.html">EC2 user data</a> functionality - to run an startup script that will configure Ghost
+as soon as the instance is started - and <a href="https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html">EC2 instance metadata</a> - to retrieve
+the public IP of the instance, and assign it dynamically to Ghost installation script._
+
+```sql TheButton[Install Ghost]="Install Ghost"
+ -- Install modules
+SELECT
+  *
+FROM
+  iasql_install ('aws_ec2', 'aws_ec2_metadata');
+
+-- insert security groups
+INSERT INTO
+  security_group (description, group_name)
+VALUES
+  ('Ghost security group', 'ghost_sg');
+
+INSERT INTO
+  security_group_rule (is_egress, ip_protocol, from_port, to_port, cidr_ipv4, description, security_group_id)
+SELECT
+  t.is_egress,
+  t.ip_protocol,
+  t.from_port,
+  t.to_port,
+  t.cidr_ipv4,
+  t.description,
+  security_group.id
+FROM
+  security_group,
+  (
+    VALUES
+      (FALSE, 'tcp', 22, 22, '0.0.0.0/0', 'ghost_sg_group_rule_ssh'),
+      (FALSE, 'tcp', 80, 80, '0.0.0.0/0', 'ghost_sg_group_rule_http'),
+      (FALSE, 'tcp', 443, 443, '0.0.0.0/0', 'ghost_sg_group_rule_https'),
+      (TRUE, 'tcp', 1, 65535, '0.0.0.0/0', 'ghost_sg_group_rule_egress')
+  ) AS t (is_egress, ip_protocol, from_port, to_port, cidr_ipv4, description)
+WHERE
+  security_group.group_name = 'ghost_sg_group';
```

Comment:
I think since you followed my suggestion in the security group name this has to change to:
```suggestion
  security_group.group_name = 'ghost_sg';
```